### PR TITLE
* 1240 feature request new fade inout ability for kryptonforms

### PR DESCRIPTION
**Pull request by Wagnerp**
* Implement #1240 
   - The developer needs to enable this, as it's turned off by default
   - Default timing values may need tinkering with

image

### CODE REVIEW

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualForm.cs [hunk 1]
```diff
@@ -400,6 +400,17 @@ public PaletteMode PaletteMode
...
+        /// <summary>Gets access to the fade values.</summary>
+        [Category(@"Visuals")]
+        [Description(@"Form fading.")]
+        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
+        public FadeValues FadeValues { get; } = new FadeValues();
+
+        private bool ShouldSerializeFadeValues() => !FadeValues.IsDefault;
+
+        /// <summary>Resets the fade values.</summary>
+        public void ResetFadeValues() => FadeValues.Reset();
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualForm.cs:412]: `Reset####`s should be `private`.

File: Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        #endregion
+
+        #region Public
+
+        /// <summary>Gets or sets a value indicating whether [fading enabled].</summary>
+        /// <value><c>true</c> if [fading enabled]; otherwise, <c>false</c>.</value>
+        [Category(@"Data")]
+        [Description(@"Enables the fading effects. Default is 'false'.")]
+        [DefaultValue(DEFAULT_FADING_ENABLED)]
+        public bool FadingEnabled { get; set; }
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs:40]: All these designer values need `ShouldSerialise###` and `Reset###` functions to work correctly in designers.

File: Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                                          ShouldCloseOnFadeOut.Equals(DEFAULT_SHOULD_CLOSE_ON_FADE_OUT) &&
+                                          FadeSpeed.Equals(DEFAULT_FADE_SPEED) &&
+                                          ParentForm!.Equals(DEFAULT_PARENT_FORM) &&
+                                          NextWindow!.Equals(DEFAULT_NEXT_FORM);
+
+        #endregion
+
+        #region Implementation
+
+        public void Reset()
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs:103]: Make this `internal`

File: Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        {
+            Reset();
+        }
+
+        #endregion
+
+        #region IsDefault
+
+        [Browsable(false)]
+        public override bool IsDefault => FadingEnabled.Equals(DEFAULT_FADING_ENABLED) &&
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs:92]: Make this `internal`

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonToastNotificationManager.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ *  Modifications by Peter Wagner(aka Wagnerp) & Simon Coghlan(aka Smurf-IV), et al. 2023 - 2023. All rights reserved.
+ *
+ */
+#endregion
+
+namespace Krypton.Toolkit
+{
+    [ToolboxItem(true)]
+    [ToolboxBitmap(typeof(KryptonToastNotificationManager), @"ToolboxBitmaps.KryptonInputBox.bmp")]
+    public sealed class KryptonToastNotificationManager : Component
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonToastNotificationManager.cs:14]: Do not make them `sealed`, as this prevents others from deriving and adding extra functionality (i.e. Ext code base)

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualBasicToastForm.cs [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+        /// <param name="showCloseButton">The show close button.</param>
+        /// <param name="customSoundStream">The custom sound stream.</param>
+        /// <param name="systemSound">The system sound.</param>
+        /// <param name="actionButtonCommand">The action button command.</param>
+        /// <param name="toastNotificationIcon">The toast notification icon.</param>
+        /// <param name="customIcon">The custom icon.</param>
+        /// <param name="inputControlStyle">The input control style.</param>
+        /// <param name="actionType">Type of the action.</param>
+        /// <param name="actionButton">The action button.</param>
+        public VisualBasicToastForm(string notificationContentText, string? notificationTitleText,
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualBasicToastForm.cs:83]: Why is this not a structure parameter from the "get go"

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualForm.cs (hunk 1 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualForm.cs:412]: Done

File: Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs (hunk 3 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs:103]: Done

File: Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs (hunk 2 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs:40]: Done

File: Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs (hunk 4 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Values/FadeValues.cs:92]: Not possible to do, as it derives from `Storage`

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonToastNotificationManager.cs (hunk 5 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonToastNotificationManager.cs:14]: Done

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualBasicToastForm.cs (hunk 6 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualBasicToastForm.cs:83]: Going to test this with fields for now, then implement the existing structure at a later date, until I'm sure that everything is functioning properly.

**Comment by Wagnerp** [comment]
> Where is the equivalent update to the Forms Demo ? In the attached gif, it appears to fade in and then jump at the last moment - That's not nice.

There is a bug somewhere, not sure if it's something to do with timings

**Pull request merged**